src/libstd/sys/windows/fs.rs

```diff
@@ -310,6 +313,43 @@ impl File {
         }
     }
 
+    #[cfg(target_os = "uwp")]
+    pub fn file_attr(&self) -> io::Result<FileAttr> {
+        unsafe {
+            let mut info: c::FILE_BASIC_INFO = mem::zeroed();
+            let size = mem::size_of_val(&info);
+            cvt(c::GetFileInformationByHandleEx(self.handle.raw(),
+                                              c::FileBasicInfo,
+                                              &mut info as *mut _ as *mut c_void,
+                                              size as c::DWORD))?;
+            let mut attr = FileAttr {
+                attributes: info.FileAttributes,
+                creation_time: c::FILETIME {
+                    dwLowDateTime: info.CreationTime as c::DWORD,
+                    dwHighDateTime: (info.CreationTime >> 32) as c::DWORD,
+                },
+                last_access_time: c::FILETIME {
+                    dwLowDateTime: info.LastAccessTime as c::DWORD,
+                    dwHighDateTime: (info.LastAccessTime >> 32) as c::DWORD,
+                },
+                last_write_time: c::FILETIME {
+                    dwLowDateTime: info.LastWriteTime as c::DWORD,
+                    dwHighDateTime: (info.LastWriteTime >> 32) as c::DWORD,
+                },
+                file_size: 0,
+                reparse_tag: 0,
+            };
+            cvt(c::GetFileSizeEx(self.handle.raw(), &mut attr.file_size as *mut _ as *mut i64))?;
```

Comment:
Not sure if Windows Store Apps are targeted here, but here is what MSDN has to say:

https://docs.microsoft.com/en-us/windows/desktop/api/fileapi/nf-fileapi-getfilesizeex#remarks

> Windows Store apps:  GetFileSizeEx is not supported. Use GetFileInformationByHandleEx.

Reply:
Oh yeah good point, I went straight down to the "minimum supported client" part... 
I'll replace it with `GetFileInformationByHandleEx` then. Thanks for noticing!

Reply:
Done!